modules/datastore/src/Controller/QueryController.php

```diff
@@ -214,10 +214,17 @@ private function sendRows($handle, array $data) {
   }
 
   /**
-   * Add the header row.
+   * Add the header row, from specified properties, if any, or the schema.
    */
   private function addHeaderRow(array &$data) {
-    $header_row = array_keys(reset($data['schema'])['fields']);
+
+    if (!empty($data['query']['properties'])) {
+      $header_row = $data['query']['properties'];
```

Comment:
@thierrydallacroce possible to provide test coverage for this condition?

Reply:
Sure, trying

Reply:
Here you go, should be good.

<img width="1785" alt="Screen Shot 2021-07-20 at 5 28 13 PM" src="https://user-images.githubusercontent.com/729791/126411684-9068564f-02c3-44a4-a974-0ab8fe5fd2cd.png">

Reply:
Let me know if you have any questions or concerns.